README.md

```diff
@@ -1,148 +1,415 @@
+* Iptables     - Although not explicitly required, a good firewall should be
+                 installed.
+
+If you are running Debian or Ubuntu, you can make sure that these are
+installed by running:
+
+    sudo apt-get install default-jre perl python wget rdiff-backup socat iptables
+    
+### Configuring the firewall / NAT
+If you have a firewall installed on your computer, or a router using NAT
+installed in your network, you will need to route some ports to your server.
+Instructions on how to accomplish this are beyond the scope of this post, but
+here are some things you will need to know:
+* The default port for the Minecraft server is: `25565`.
+* If you wish to run multiple world servers using this script, you will
+  want to open a range of ports (for example `25565 - 25575`).
+
+See the [iptables.rules](iptables.rules)
+file for a very basic set of rules that you can use with the Iptables firewall.
+
+### Mapping software (optional)
+The script uses the [Minecraft Overviewer](http://overviewer.org) mapping
+software to generate maps of your worlds.  Minecraft Overviewer is a
+command-line tool for rendering high-resolution maps of Minecraft worlds. It
+generates a set of static html and image files and uses the Google Maps API to
+display a nice interactive map.
 
+If you wish to use the mapping software, you can [download]
+(http://overviewer.org/downloads) premade binaries for
+supported systems, or build your own binary from source if needed.
+
+Repositories for automatic installation are also available:
+* [Debian/Ubuntu](http://overviewer.org/debian/info)
+* [RHEL/CentOS/Fedora](http://overviewer.org/rpms/info)
 
 ## Installation
+### Downloading the script
+The easiest way to download the script is to make a clone of the [git
+repository](https://github.com/sandain/MinecraftServerControlScript.git). 
+You must have git installed first. To install git:
+
+    sudo apt-get install git
+        
+Then:
+
+    git clone https://github.com/sandain/MinecraftServerControlScript.git
+        
+Note that it will be downloaded into the current directory which you are working
+ in. 
 
-### Download
-You can download the script from the following locations:
+##### Other ways to download
 
-* Get the latest stable [release](https://github.com/sandain/MinecraftServerControlScript/releases).
+* Get the latest stable [release]
+(https://github.com/sandain/MinecraftServerControlScript/releases).
 
-* Get the development version as a [zip file](https://github.com/sandain/MinecraftServerControlScript/archive/master.zip):
+* Get the development version as a [zip file]
+(https://github.com/sandain/MinecraftServerControlScript/archive/master.zip):
 
-        wget https://github.com/sandain/MinecraftServerControlScript/archive/master.zip
 
-* Make a clone of the [git repository](https://github.com/sandain/MinecraftServerControlScript.git):
+     wget https://github.com/sandain/MinecraftServerControlScript/archive/master.zip
```

Comment:
Indenting this doesn't seem to put it in a code block, could be the markdown parser thinks it's a new line for the list item above. Might have to wrap this in ```